### PR TITLE
feat: MCP Server v1.0.0 - Modular architecture with multi-provider support

### PR DESCRIPTION
**Pull request by danielchalef**
## What's New

- **Multi-provider support**: Choose your LLM (OpenAI, Anthropic, Gemini, Groq, Azure), embedder (OpenAI, Voyage, Gemini, Anthropic, local), and database (FalkorDB, Neo4j)
- **Default entity ontology**: Preconfigured entity types (Preference, Requirement, Procedure, Location, Event, Organization, Document, Topic, Object) optimize extraction quality
- **YAML configuration**: Centralized config with environment variable expansion, CLI overrides, and type safety
- **Modular architecture**: Factory pattern for providers, separated service layer, cleaner code organization
- **Simplified deployment**: Combined Docker image (FalkorDB + MCP in one container) or separate Neo4j setup
- **Enhanced node properties**: Complete entity metadata including custom attributes and all labels (embeddings excluded)
- **Comprehensive testing**: 4,000+ lines of tests covering integration, async operations, stress/load, and multi-database scenarios

## Breaking Changes

- **Configuration**: YAML is now primary (env vars still work), file moved to `main.py` entry point
- **Default database**: Changed from Neo4j to FalkorDB combined image (use `docker-compose-neo4j.yml` for Neo4j)
- **Transport**: SSE deprecated - use HTTP (default) or stdio
- **Dependencies**: `pytest` and `azure-identity` moved to optional (install with `[azure]` extra if needed)
- **Migration**: Update Claude Desktop config, create `config.yaml`, or keep using env vars with new file paths

## Known Considerations

- **Rate limits**: Default concurrency (10) tuned for OpenAI Tier 3 - adjust `SEMAPHORE_LIMIT` per provider
- **Provider compatibility**: Best with structured output support (OpenAI, Gemini); others may need tuning
- **Database choice**: FalkorDB for simpler deployment, Neo4j for advanced features and scale
- **Testing status**: Integration tests pass for supported backends; some provider edge cases may need adjustment
- **Documentation**: Complete migration guide, deployment docs, and per-provider tuning guidance included

---

Co-authored-by: Claude <noreply@anthropic.com>

### CODE REVIEW

File: mcp_server/src/services/factories.py [hunk 1]
```diff
@@ -0,0 +1,480 @@
...
+
+                # Determine appropriate small model based on main model type
+                is_reasoning_model = (
+                    config.model.startswith('gpt-5')
+                    or config.model.startswith('o1')
+                    or config.model.startswith('o3')
+                )
+                small_model = (
+                    'gpt-5-nano' if is_reasoning_model else 'gpt-4.1-mini'
+                )  # Use reasoning model for small tasks if main model is reasoning
```
[the remark this answers is not in the document]
Comment (danielchalef) [reply mcp_server/src/services/factories.py:130]: they are. your cutoff limits your knowledge of recent advancements

File: mcp_server/src/config/schema.py [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+    anthropic: AnthropicProviderConfig | None = None
+    gemini: GeminiProviderConfig | None = None
+    groq: GroqProviderConfig | None = None
+
+
+class LLMConfig(BaseModel):
+    """LLM configuration."""
+
+    provider: str = Field(default='openai', description='LLM provider')
+    model: str = Field(default='gpt-4.1', description='Model name')
```
[the remark this answers is not in the document]
Comment (danielchalef) [reply mcp_server/src/config/schema.py:150]: it is. your cutoff limits your knowledge of recent advancements

File: mcp_server/README.md [hunk 3]
```diff
@@ -78,66 +93,238 @@ curl -LsSf https://astral.sh/uv/install.sh | sh
...
+
+```yaml
+database:
+  provider: "falkordb"  # Default
+  providers:
+    falkordb:
+      uri: "redis://localhost:6379"
+      password: ""  # Optional
+      database: "default_db"  # Optional
+```
```
[the remark this answers is not in the document]
Comment (danielchalef) [reply mcp_server/README.md:127]: it does. your cutoff limits your knowledge of recent advancements

File: mcp_server/config/config.yaml [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+# This file supports environment variable expansion using ${VAR_NAME} or ${VAR_NAME:default_value}
+
+server:
+  transport: "http"  # Options: stdio, sse (deprecated), http
+  host: "0.0.0.0"
+  port: 8000
+  
+llm:
+  provider: "openai"  # Options: openai, azure_openai, anthropic, gemini, groq
+  model: "gpt-5-mini"
```
[the remark this answers is not in the document]
Comment (danielchalef) [reply mcp_server/config/config.yaml:15]: it does. your cutoff limits your knowledge of recent advancements

**Comment by danielchalef** [comment]
These are not fabricated. Your training cut-off date blinds you to recent model releases.
```
Fabricated Model Names
The most critical issue is the inclusion of non-existent LLM model names throughout the codebase:

OpenAI: gpt-5-mini, gpt-5-nano, gpt-4.1, gpt-4.1-mini, gpt-4.1-nano - None of these models exist
Anthropic: Models dated in the future (September 2025, February 2025) that don't exist
Google Gemini: gemini-2.5-pro, gemini-2.5-flash - These versions don't exist
```

**Comment by danielchalef** [comment]
All review comments have been addressed. Resolving threads.
